runtime/j9vm/jvm.c

```diff
@@ -1370,157 +1355,49 @@ decodeSetting(const char* key, const char* value, const VersionSetting* settings
 }
 
 /**
- * Attempt loading 'classlib.properties' file, and get Java version info.
- * If the file is found, 'version' and 'shape' values are retrieved
- * and decoded as J2SE_xx and J2SE_SHAPE_xx accordingly.
- * 'J2SE_xx | J2SE_SHAPE_xx' is returned;
- * Otherwise, 0 is returned.
+ * Get Java version of running JVM
+ * Attempt getting the Java version info from classlib.properties,
+ * if the file can't be opened, J2SE_LATEST is returned;
+ * if the file is opened successfully, either J2SE_18 or a version decoded 
+ * from the version info within the file is returned.
  *
- * @return 'J2SE_xx | J2SE_SHAPE_xx' decoded from 'version' and 'shape' values in 'classlib.properties';
- *         or 0 if otherwise.
+ * @return J2SE_xx decoded from classlib.properties,
+ *         or J2SE_18 if classlib.properties exists but no version decoded,
+ *         or 'J2SE_LATEST'.
  */
 static UDATA
-getVersionFromClasslibPropertiesFile(void)
+getVersionFromPropertiesFile(void)
 {
-	PORT_ACCESS_FROM_PORT(&j9portLibrary);
-	J9StringBuffer *propsPathBuffer = NULL;
-	j9props_file_t propsFile = NULL;
-	UDATA finalVersion = 0;
-
-	propsPathBuffer = jvmBufferCat(propsPathBuffer, jvmBufferData(j9libBuffer));
-	propsPathBuffer = jvmBufferCat(propsPathBuffer, DIR_SEPARATOR_STR "classlib.properties");
-	propsFile = props_file_open(PORTLIB, jvmBufferData(propsPathBuffer), NULL, 0);
-	free(propsPathBuffer);
-	propsPathBuffer = NULL;
-
-	if (NULL != propsFile) {
-		const char *shape = NULL;
-		const char *version = NULL;
-		UDATA decoded = 0;
-		
-		shape = props_file_get(propsFile, "shape");
-		if (NULL == shape) {
-#ifdef DEBUG
-			printf("No 'shape' property in %s\n", jvmBufferData(propsPathBuffer));
-#endif
-			goto bail;
-		}
-
-		version = props_file_get(propsFile, "version");
-		if (NULL == version) {
-#ifdef DEBUG
-			printf("No 'version' property in %s\n", jvmBufferData(propsPathBuffer));
-#endif
-			goto bail;
-		}
+	if (-1 == jvmSEVersion) {
+		PORT_ACCESS_FROM_PORT(&j9portLibrary);
+		J9StringBuffer *propsPathBuffer = NULL;
+		j9props_file_t propsFile = NULL;
+		UDATA finalVersion = J2SE_18;
```

Comment:
Why this default? I don't see this behavior in the original code.

Reply:
When there is a `classlib.properties`, default version is `J2SE_18` for `Java 8` which is the only build with this file.
Existing code sets the `finalVersion` to `J2SE_LATEST` if there is an error.
I thought this makes it is explicit that only non-openj9 `Java 8` build has this property file.

Reply:
I think it makes more sense to set J2SE_LATEST as the default. Just because a classlib.properties file is found doesn't mean it should default to J2SE_18 when the file doesn't contain a recognized version.  Defaulting to J2SE_LATEST is how it works now. J2SE_LATEST will be J2SE_18 for any Java 8 build anyway.

Also, I'm fine to have J2SE_LATEST as the initialization value for finalVersion, it doesn't need to be set in all the cases.

Reply:
> J2SE_LATEST will be J2SE_18 for any Java 8 build anyway.

It seems there is no need to read/parse `classlib.properties`, i.e., always use `J2SE_LATEST`. The property file can be deleted.

Reply:
We can't delete the property file, it is used for other things like the bootclasspath. However good point, we don't need to read the version.

Reply:
PR has been updated, please review.